lvsfunc/comparison.py

```diff
@@ -247,12 +249,12 @@ def _compare(self) -> vs.VideoNode:
         return core.std.StackVertical(rows)
 
     def _auto_arrangement(self) -> List[List[int]]:
-        def _grouper(iterable, n, fillvalue=None):  # type: ignore
+        def _grouper(iterable: Iterable[T], n: int, fillvalue: Optional[T] = None) -> Iterator[Tuple[T, ...]]:
             args = [iter(iterable)] * n
             return zip_longest(*args, fillvalue=fillvalue)
 
         dimension = 1 + math.isqrt(self.num_clips - 1)
-        return list(map(list, _grouper([1] * self.num_clips, dimension, 0)))
+        return list(map(lambda x: list(x), _grouper([1] * self.num_clips, dimension, 0)))
```

Comment:
if `map(list, …)` doesn't work for some reason, better use a comprehension. You'll almost never want to use a lambda inside a map.

```suggestion
        return [list(x) for x in _grouper([1] * self.num_clips, dimension, 0)]
```

Also note that for a clip size of 20, for example, you'll end up with a dimenion of 5 but because 20 is divided by 5, the resulting clip's aspect ratio will change. I thought this was unintended, which is why I wrote my suggestion differently (and without `fillvalue`).

Reply:
I don't think the aspect ratio was meant to be guaranteed to stay the same, maybe I could put that as an option? But realistically you can specify your own square arrangement if you'd want to do that. As for the list comprehension, yea that seems a bit nicer to read.

Reply:
Actually, I quite like the way it handles dropping blank rows now in the auto_arrangement because it results in the most optimal (spacial-wise and filesize if you were to export it as an image) way of tiling the clips. I don't think anyone would be encoding something like this as this would be *very* ram intensive and not sure what resolutions common encoders even support.